agents/holo_fpga/roach_agent.py

```diff
@@ -0,0 +1,188 @@
+        # so does not require the lock
+        self.initialized = True
+
+        return True, "FPGA connected."
+
+    def take_data(self, session, params=None):
+
+        with self.lock.acquire_timeout(timeout=3, job="take_data") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not set position because lock held by {self.lock.job}"
+                )
+                return False, "Could not acquire lock"
+
+            # Grab synthesizer settings here (which FPGA bins to integrate over)
+            self.synth_settings = synth3.SynthOpt()
+
+            self.synth_settings.IGNORE_PEAKS_BELOW = int(738)
+            self.synth_settings.IGNORE_PEAKS_ABOVE = int(740)
+            # Take data here
+            arr_aa, arr_bb, arr_ab, arr_phase, arr_index = fpga_daq3.TakeAvgData(
+                self.baseline, self.fpga, self.synth_settings
+            )
+
+            # Data dictionary is what we will send to the data feed:
+            data = {"timestamp": time.time(), "block_name": "fpga", "data": {}}
+
+            arr_AA = np.array(fpga_daq3.running_mean(arr_aa.tolist(), 1))
+            arr_BB = np.array(fpga_daq3.running_mean(arr_bb.tolist(), 1))
+            arr_AB = np.array(fpga_daq3.running_mean(arr_ab.tolist(), 1))
+            arr_P = np.array(fpga_daq3.running_mean(arr_phase.tolist(), 1))
+
+            n_channels = np.size(arr_AA)
+
+            amp_AA = arr_AA[int(n_channels / 2)]
+            amp_BB = arr_BB[int(n_channels / 2)]
+            amp_AB = np.power(arr_AB[int(n_channels / 2)], 1)
+            amp_P = np.remainder(arr_P[int(n_channels / 2)], 360.0)
+
+            data["data"]["amp_AA"] = amp_AA
+            data["data"]["amp_BB"] = amp_BB
+            data["data"]["amp_AB"] = amp_AB
+            data["data"]["amp_P"] = amp_P
+
+            self.agent.publish_to_feed("fpga", data)
+            session.data.update(data["data"])
+
+            pass
+
+        return True, "Data acquired."
+
+
+def make_parser(parser=None):
+    """Build the argument parser for the Agent. Allows sphinx to automatically
+    build documentation based on this function.
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    # Add options specific to this agent.
+    pgroup = parser.add_argument_group("Agent Options")
+    pgroup.add_argument("--config_file")
+
+    return parser
+
+
+if __name__ == "__main__":
+    # For logging
+    txaio.use_twisted()
+    LOG = txaio.make_logger()
+
+    # Start logging
+    txaio.start_logging(level=os.environ.get("LOGLEVEL", "info"))
+
+    parser = make_parser()
+
+    # Interpret options in the context of site_config.
+    args = site_config.parse_args(agent_class="FPGAAgent", parser=parser)
+
+    agent, runner = ocs_agent.init_site_agent(args)
+
+    fpga_agent = FPGAAgent(agent, args.config_file)
+
+    agent.register_task("init_FPGA", fpga_agent.init_FPGA)
```

Comment:
Can we PEP08-ify `init_FPGA` to `init_fpga`? (I'm mainly concerned with the task name here, as that forms the API for the Agent. Changing the method name `fpga_agent` is only internal to this agent.)

Reply:
Removed this function all together, but I'll remember for future functions!